src/d2s.rs

```diff
@@ -85,21 +85,27 @@ fn mul_shift_all(
     let lo2 = lo.wrapping_add(mul.0);
     let mid2 = mid.wrapping_add(mul.1).wrapping_add((lo2 < lo) as u64);
     let hi2 = hi.wrapping_add((mid2 < mid) as u64);
-    *vp = shiftright128(mid2, hi2, j - 64 - 1);
+    unsafe {
+        ptr::write(vp, shiftright128(mid2, hi2, j - 64 - 1));
```

Comment:
This function is safe, so it should not write to arbitrary pointers.

Reply:
Thanks! Good catch.